PathOfBuildingCommunity/PathOfBuilding
# Add Atlas Tree support

WIP. Doesn't crash

- Import / Export and Multiple atlas trees.
- No compare yet.
- Lots of commented code. Tried not to delete too much as I wasn't sure what needed to come back after i stopped the app crashing.
- Would like the stat summary to appear in the stat box on the left.
- The more nodes selected, the longer it take to select another.

Questions
- Do you want this ?
- Do the Skills / Items / Calcs buttons need spreading out to fill line

Fixes #4033.

### Description of the problem being solved:
GGG introduced the atlas tree.

### Steps taken to verify a working solution:
- Open a build
- Backup the xml file for the build you have open (just a confidence thing, for you that is) - you may have a good diff'ing tool that you want to use.
- Click on 'Atlas Tree'.
- Select nodes to create a tree.
- Click on 'Export Tree', copy it and paste it into a browser. Confirm you have the same tree.
- Click on 'Manage Trees ...'
- Create a new Atlas tree and make it active.
- Click on 'Import Tree'. Tree should be the same.
- Select the tree select control, hover over the default tree :
![image](https://user-images.githubusercontent.com/4302241/154800744-b8231205-1ca8-4e63-8978-73d9458cb6f1.png)
- Add some more points to your second tree.
- Select the tree select control, hover over the default tree :
![image](https://user-images.githubusercontent.com/4302241/154800784-a8f46741-32d0-4c08-af05-5d14a10d58e8.png)
- Save the build
- Open the build's .xml in an editor. Note the entry '<Atlas activeSpec="2">'. URLs are correct PoE urls.
- Back in PoB, close the build.
- Open the build again. Check that the Atlas tab is default (cause i didn't say to change it did I ?). The second tree is active.
- In a browser, open https://poeskilltree.com/?v=3.17.0-atlas
- Create a tree. Copy the url
- Back in PoB, start a new build.
- Select 'Import tree' from the 'Atlas Tree page'. The same tree should be shown as is shown in the browser.
- Select 'Import/Export Build' and use the pastebin/code provided below. You will get the two trees use in the hover examples above.
- What else ?

### Link to a build that showcases this PR:
https://pastebin.com/mT9dSG0f
Build code in case pastebin misbehaves :
<pre>eNrFWt9z4jYQfs79FR6eScA_MNCBdgghCZ2Qo5jcXfuSUYzAusgWtWU47q_vSobApZCsdZ1pHhLjfKvd_bS72lXS-e1bzK0VTTMmkm7FvqhXLJqEYsaSRbfyML0-b1V--_VDZ0xk9HF-mTOufvPrh7OOfrY4XVEOchVLknRB5afdSu4jrLQkiYyoSEbkq0hvxKxbuRcJrVhPJJkxufsUcpJl9ySm3UoQgnDFIllIk1l__74AxoQlgQifqbxJRb7UaleMrkdiBpjhaPxxMj1QypJDpWDzWWfMyYamgSTSyuBbt9ID18mCXpEYvsNqhOdU0dBqNZq25-jvbqV2XDhYUjo7EHJOAccpHcznNJRsRfspk_2IJOFeXf2UXFnsKOeSLTmj6YFVjVMSt_9avN0-hZ0KSfjVOHiBOheO7-xJeltOyPft_8xkdMmBUAMtSna4SJikhsJjwTKR_IR_h6KntyjnHLIHhxXxE0uM2BiRhPRFhuD8js0pDjkIcLgJpBEOqawc0xQSXeKNLSWw1RDQUEAxKaOjjMiBH2bKDCQHgYmAgaJA7iuJcxJ1Rb8hUMNEotZaCakPgfds06k3uB2_IP3mhet77bbftuv1tn_SJ-0_nA4sJnxEvrE4j6EUTskz3Wt1m6frptq3vdL6G6dD-IvCDpMQFwwPSUozmq4ODhUfKbDd3n3Zr78pOKELmuCMuqM0jG7g5J0QSXHp8IJqvE2OwqLIUcAj5DSQAiXIUYKvyLlovwUuSc9nks4Q6ZrQdLEJIkY5Aq25PBRBcXoogAyHQ5GSfg9WJDvMatt_25sCjgsOCl0DCMwotlUap-KrasZ4ObFeGos8Re5HAUY5MI42GQuhjdBN6ITO8hBXAV96ypFY0RgiXDek0A3vaT4lesmhj8Y6DstyXkqiJyUJn6_EbEFLKSkvEeTLJeS5ihas3DVLgeWMHRyA5z4C_RFmhT5ZYno3lVJYBXs0WsEdW0QygVYSr-WVCN6XiIishDN7OFrFNbTFqJ54LNawZKQGzawcGo75_dF-0o6UJt836PV_gKMUDJJZnqoYRet4LfFvNZ2aHsTV0zBeilTqlz0Js7NFdHFQ5Qg6rmIAVh8Ox-shjMcJ42q2ziRNN0VJybqVipXAVA0PjtdoNKqO59fbVadd9xrVhufVm1XfdW276nqO51VhLGm7VdtpuX7Vabb8etVtglzFkimlB1cCdnM77Su1dW3RWWcwg7ltdq-01Yo3D5M7_XAWSbnMfqnV1uv1xZLISMzpN6jaF6GIa0tYBrw7z54Z5-dKUa0HX5eLXq__6TH9s3_7ED3OxlE4ieV0-Px0H0zmf_R6WkFtp6FTXCcUijs1xc6eJskkB56na1EpR5ndaPvNakGc79W91hH6Gs2m296R5TYbpwm1662W9_8SevW1Mf9yvvJHN73IHs4j5nx_WLRdd-p9GfyVPmx-7_89eP5SB3S3iyG4U9MBqp6moOeHQLVPB2r9Dc4brZbr_l_8qC-c38pd9XAvJFX-n6mXuw-dQC2dWRmk8Q2Ns8sNlMVr1W29ug-IiYTmi8Z36vZtKlRxAAppelfcxs0Jz-Ac_GGZbkWmObyc0TnJuXr_R044kxtN6_7tdgkY0KwsEmt1vBbLTDdLxXPv7q74TY_L7QpKx06prj59wkPtXGeYLHNpJfoOL2ZZ-PiUz-fqug7WkKm-YxxcXw_60-GnwbYGHopoqh-TPH5SN1rFz31TE1DdJ1mh4JwsMzp78ZzBo-YygN4wlCg8jIDbW0AMGvr4MGVQwjFgdf-FMlnfJ2KQ6rIPBYQzgXCKJ0ztsjp_Mixn26YLg9a3eyhkcR2Hc1Bf--G2gYZkg0K-DDco9IDTHuOq_8axNoJE2BYwFB6a6ZQ95RIZbfp6AmWHGtVRDh6Oo8jswAG3IxXKiO0EicFu-1VcUEKhRm_bFZ1TCHdk2utyUtwrFa3arjB2-iKZs0VtdwB-YnRtZRRqShTIVJ1q34WI_9SlWdXaW0rkiCxfVle_3RZqd1unIWevGARVqu3bVXsF_LLrLztDCZV6e9iq54BK_TeUPKPFpdhnSpYi0a8LTUVFLqBvwLS79vZ4DbjY1e8CaNnBmiyt3tMmywi3ipPRgmYGEjcuDubaSdHXYg5O7G3UpRAyMzTomC_vGHVJuSwpcg1Ny7PlokxyjpnkmtGLI8H5j8Q8MyN9M22-OZsm8fTODtxSODJkWUVHIsk30fNO-N1wsVLzafmFXfMssk30eWaxYJtFnvf-9mCILbvssfAydBxVdxoG9hjUQPc9NbONVXQIJbfvOMtG8dUwEUKlJGYjjPxsGNPqGOy7b5ZJrln4uuWjzDMv-wYxbRuT72ICwsOADMz-ieYHuZPmu-AbRKX5NvxEuGBLspJFZg0easDchCULXCdQulCcpNewXOD308QdF8GTbb4NnjFbDRNvMPliHuTvmNSLc37kbOvUtlOknoP1IPryd4qT06_r-D_MujZm1j1v6n_c6NRe_8viP-Fas5w=</pre>

### Before screenshot:
![image](https://user-images.githubusercontent.com/4302241/154800466-7d1375cf-87fc-4c59-9b7f-04987d472a4c.png)

### After screenshot:
![image](https://user-images.githubusercontent.com/4302241/154800481-a130cef1-3491-4833-a4c5-e450b03ed87c.png)
![image](https://user-images.githubusercontent.com/4302241/154800494-03622acb-7c4e-42f7-9049-903737dcaf26.png)
